usr.sbin/ctld/ctld.h

```diff
@@ -49,25 +57,34 @@
 #define	MAX_LUNS			1024
 #define	SOCKBUF_SIZE			1048576
 
+struct isns_req;
+struct port;
+
 struct auth {
-	TAILQ_ENTRY(auth)		a_next;
-	struct auth_group		*a_auth_group;
-	char				*a_user;
-	char				*a_secret;
-	char				*a_mutual_user;
-	char				*a_mutual_secret;
-};
+	auth(const char *secret) : a_secret(secret) {}
+	auth(const char *secret, const char *mutual_user,
+	    const char *mutual_secret) : a_secret(secret),
+	    a_mutual_user(mutual_user), a_mutual_secret(mutual_secret) {}
+
+	bool mutual() const { return !a_mutual_user.empty(); }
 
-struct auth_name {
-	TAILQ_ENTRY(auth_name)		an_next;
-	struct auth_group		*an_auth_group;
-	char				*an_initiator_name;
+	const char *secret() const { return a_secret.c_str(); }
+	const char *mutual_user() const { return a_mutual_user.c_str(); }
+	const char *mutual_secret() const { return a_mutual_secret.c_str(); }
+
+private:
+	std::string			a_secret;
+	std::string			a_mutual_user;
+	std::string			a_mutual_secret;
 };
 
 struct auth_portal {
-	TAILQ_ENTRY(auth_portal)	ap_next;
-	struct auth_group		*ap_auth_group;
-	char				*ap_initiator_portal;
+	auth_portal(const struct sockaddr_storage *ss, int mask) : ap_sa(*ss),
+	    ap_mask(mask) {}
+
+	bool matches(const struct sockaddr_storage *ss) const;
```

Comment:
I think for method prototypes we should follow the same rule as standalone prototypes and data members in structs in C where if the return type name is short enough, a tab is used to separate the return type from the method name (so in this case it would change to use a tab here after `bool`)